src/dafny/Collections/Sets.dfy

```diff
@@ -193,4 +193,35 @@ module {:options "-functionSyntax:4"} Dafny.Collections.Sets {
     LemmaSubsetSize(x, range);
   }
 
+  /** Any totally ordered set contains a unique minimal element. */
+  lemma LemmaUniqueMinimum<T(!new)>(R: (T, T) -> bool, s: set<T>) returns (m: T)
```

Comment:
Shouldn't you have a lemma that proves that for any set, even partially ordered ones, there is at most one minimum, and a separate lemma that states that for a totally ordered set, there is at least one minimum?

Reply:
In my opinion, it is fine to capture uniqueness in the post-condition of `LemmaUniqueMinimum`, as it is done right now: `ensures IsMinimum(R, m, s) && (forall n: T | IsMinimum(R, n, s) :: m == n)`.

Reply:
I think it's fine too. But just to be clear, proving that for general relations, without a total ordering constraint, there is at most one minimum, gives a more generally applicable lemma, right?

Reply:
True, to prove uniqueness, anti-symmetry should be sufficient.

Reply:
I added separate uniqueness proofs.